tests/local/sql/test_filter_and_group_by.py

```diff
@@ -9,89 +9,18 @@
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
 import logging
-from typing import List
 
-from sodasql.scan.metric import Metric
-from sodasql.scan.scan_column import ScanColumn
-from sodasql.scan.scan_yml_parser import ScanYmlParser
-from sodasql.scan.warehouse import Warehouse
 from tests.common.sql_test_case import SqlTestCase
 
 
-def execute_metric(warehouse: Warehouse, metric: dict, scan_dict):
-    dialect = warehouse.dialect
-
-    scan_configuration_parser = ScanYmlParser(scan_dict, 'Test scan')
-    scan_configuration_parser.assert_no_warnings_or_errors()
-    scan = warehouse.create_scan(scan_configuration_parser.scan_yml)
-    scan.execute()
-
-    fields: List[str] = []
-    group_by_column_names: List[str] = metric.get('groupBy')
-    if group_by_column_names:
-        for group_by_column in group_by_column_names:
-            fields.append(dialect.qualify_column_name(group_by_column))
-
-    column_name: str = metric.get('columnName')
-    qualified_column_name = dialect.qualify_column_name(column_name)
-
-    type = metric['type']
-    if type == Metric.ROW_COUNT:
-        fields.append('COUNT(*)')
-    if type == Metric.MIN:
-        fields.append(f'MIN({qualified_column_name})')
-    elif type == Metric.MAX:
-        fields.append(f'MAX({qualified_column_name})')
-    elif type == Metric.SUM:
-        fields.append(f'SUM({qualified_column_name})')
-
-    sql = 'SELECT \n  ' + ',\n  '.join(fields) + ' \n' \
-          'FROM ' + scan.qualified_table_name
-
-    where_clauses = []
-
-    filter = metric.get('filter')
-    if filter:
-        where_clauses.append(dialect.sql_expression(filter))
-
-    scan_column: ScanColumn = scan.scan_columns.get(column_name)
-    if scan_column and scan_column.non_missing_and_valid_condition:
-        where_clauses.append(scan_column.non_missing_and_valid_condition)
-
-    if where_clauses:
-        sql += '\nWHERE ' + '\n      AND '.join(where_clauses)
-
-    if group_by_column_names:
-        sql += '\nGROUP BY ' + ', '.join(group_by_column_names)
-
-    return warehouse.sql_fetchall(sql)
-
-
```

Comment:
Where did this logic go?  Or did you remove it?  I do still see usage for it.

Reply:
You are talking about `execute_metric()`, right? I moved this to `SqlTestCase` (the same place as `scan()`).
I just thought it belong there (so it could be used in other test files). Are you ok with that?

Reply:
I moved it to `SqlTestCase` (as a class method).